patches/PSP.cmake

```diff
@@ -1,34 +1,64 @@
-SET(CMAKE_SYSTEM_NAME Generic)
-SET(CMAKE_SYSTEM_VERSION 1)
-SET(CMAKE_CROSSCOMPILING TRUE)
+#
+# CMake toolchain file for PSP.
+#
+# Copyright 2019 - Wally
+# Copyright 2020 - Daniel 'dbeef' Zalega
 
-# set compiler
-set(CMAKE_C_COMPILER psp-gcc)
-set(CMAKE_CXX_COMPILER psp-g++)
+# Determine PSP toolchain installation directory (psp-config binary is added to path after successful installation):
+execute_process(COMMAND bash -c "psp-config --pspdev-path" OUTPUT_VARIABLE PSPDEV OUTPUT_STRIP_TRAILING_WHITESPACE)
 
-SET(BUILD_SHARED_LIBS FALSE)
-SET(CMAKE_EXECUTABLE_SUFFIX ".elf")
+# Assert that PSP SDK path is now defined:
+if (NOT DEFINED PSPDEV)
+    message(FATAL_ERROR "PSPDEV not defined. Pass -DPSPDEV=/usr/local/pspdev or other installation location.")
+endif ()
 
-# set find root path
-execute_process(COMMAND psp-config --pspsdk-path
-  OUTPUT_VARIABLE PSPSDK_PATH
-  OUTPUT_STRIP_TRAILING_WHITESPACE)
+# Set helper variables:
+set(PSPSDK ${PSPDEV}/psp/sdk)
+set(PSPBIN ${PSPDEV}/bin)
+set(PSPCMAKE ${PSPDEV}/psp/share/cmake)
 
-execute_process(COMMAND psp-config --psp-prefix
-  OUTPUT_VARIABLE PSP_PREFIX
-  OUTPUT_STRIP_TRAILING_WHITESPACE)
+# Basic CMake Declarations - Required (CMAKE_SYSTEM_NAME sets cross compiling to true)
+set(CMAKE_SYSTEM_NAME Generic)
+set(TOOLCHAIN "${PSPDEV}")
+set(CMAKE_C_COMPILER ${CC})
+set(CMAKE_CXX_COMPILER ${CXX})
+set(CMAKE_FIND_ROOT_PATH "${PSPSDK};${PSPDEV}")
+set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
 
-set(CMAKE_FIND_ROOT_PATH "${PSPSDK_PATH};${PSP_PREFIX}")
-include_directories(SYSTEM "${PSPSDK_PATH}/include;${PSP_PREFIX}/include")
+# Include directories:
+include_directories(${include_directories} ${PSPSDK}/include)
```

Comment:
I think it's now no longer looking in ${PSPDEV}/psp/include

Reply:
Ah, right, mistakenly omitted this path, everything worked because this directory is empty on a vanilla toolchain, with no more dependencies installed

Reply:
Fixed